dimitriosv/week3/app/Message.jsx

```diff
@@ -0,0 +1,42 @@
+
+var Message = React.createClass({
+    eachList: function(messageId) {  return function(list,i) {
+        return (
+             <option key={list.listId+messageId} value={list.listId}>{list.listName}</option>
+        );
+    }
+    
+    },
+    render: function () {
+        return (<div>
+                    {this.props.messageIsArchived
+                        ? <div className="archived">
+                        <div className="messageText" >{this.props.messageText}</div>
+                        <button type="button" onClick={this.toggleArchiveMessageChild.bind(null,this.props.messageId)}>Unarchive</button>
+                        </div>
+                        : <div>
+                        <div className="messageText" >{this.props.messageText}</div>
+                        <button type="button"  onClick={this.deleteMessageChild.bind(null,this.props.messageId)}>Delete</button>
+                            <select className="llistcont" defaultValue="0"  onChange={this.moveMessage}>
+                                <option key="0" value="0"  >Move to:</option>
+                                {this.props.allLists.map(this.eachList(this.props.messageId))}
+                             </select>
+                        <button type="button" onClick={this.toggleArchiveMessageChild.bind(null,this.props.messageId)}>Archive</button></div>
```

Comment:
You do not need to bind `this.props.messageId`. `this.props` are class properties and can be accessed from anywhere in your class. Same applies for `deleteMessageChild` function